.github/workflows/actions/setup_dev_ubuntu/action.yaml

```diff
@@ -0,0 +1,84 @@
+name: 'Setup dev environment'
+description: 'Setup a dev environment for rgh-action Ubuntu'
+inputs:
+  version_ghc:
+    description: 'GHC version needed'
+    required: false
+    default: 8.10.7
+  version_cabal:
+    description: 'Cabal version needed'
+    required: false
+    default: 3.10.1.0
+  cache_prefix:
+    description: 'A prefix to insert to the cache id'
+    require: false
+    default: ''
+  clean_caches:
+    description: 'Should we clean old caches for this branch and keep only the last one?'
+    require: false
+    default: true
+runs:
+  using: "composite"
+  steps:
+
+    - name: "Clear old caches"
+      if: ${{ inputs.clean_caches == 'true' }}
+      shell: bash
+      run: |
+        .github/workflows/actions/setup_dev_ubuntu/clear_caches
```

Comment:
I must admit, @ch1bo, I'm not 100% sure it's needed. I'd like to see in practice but here is the rationale.

We have three caches:
* env
* cabal
* dist

env and cabal caches depend on some source files hash so they should always contain the same stuff if we don't change these files.

dist, on the contrary, depends on the workflow run. That is, every time we run the workflow in a branch (everytime we push to a PR branch) we'll generate a new list cache that will be built on top of the previous one. For instance imagine we have a PR foo:

1. push 1 to PR foo -> will fetch dist from master and generate a new dist-1 cache
2. push 2 to PR foo -> will fetch dist-1 and generate dist-2 cache
3. push 3 to PR foo -> will fetch dist-2 and generate dist-3 cache

So know we have dist-1, dist-2 and dist-3 but only dist-3 is useful, the two others will never be used again. And it happens that dist cache is 1.1Gb big and we only have 10Gb and, of course, dist-1 will be more _fresh_ than the dist from master so, maybe, I'm not sure, dist from master will be reclaimed before dist-1 which would be unfortunate.

Hence, I've implemented this clear cache feature for the useless dist caches to save space and avoid chaotique cache reclamations from GitHub.

My plan is to see in practice what happens in a realistic branch and check the output from this step to ensure it does actually save space in practice.

Reply:
Wouldn't it be better to produce the dist caches then based on a somewhat reproducible key (rather than the unique run id)? Then, at least, we have a chance for cache re-use and don't pollute caches with identical contents. For example after merging a PR, the master build should yield a cache hit and not produce a new entry if the same files have been hashed.

Reply:
Thats a good point. The only thing I can think of now would be the full code base hash but I'm not sure it's a great idea to do so. In my opinion, it would be a bit too smart, hence fragile... but maybe at some point we can lean towards that.